packages/rum-core/test/performance-monitoring/capture-hard-navigation.spec.js

```diff
@@ -76,173 +80,75 @@ describe('navigationTiming', function() {
-      { name: 'Fire "load" event', _end: 2874, _start: 2852 }
-    ])
+    const requestStartValues = [undefined, null, 0, 1, Number(new Date())]
+    for (let i = 0; i < requestStartValues.length; i++) {
+      const value = requestStartValues[i]
+      const timingObj = { ...timings, requestStart: value }
+      const spans = createNavigationTimingSpans(timingObj, transactionEnd)
+      expect(spans.map(mapSpan)).toEqual([
+        {
+          name: 'Parsing the document, executing sync. scripts',
+          _end: 1464,
+          _start: 820
+        },
+        { name: 'Fire "DOMContentLoaded" event', _end: 1498, _start: 1464 },
+        { name: 'Fire "load" event', _end: 2874, _start: 2852 }
+      ])
+    }
 
-    timings.domLoading = Number(new Date())
-    timings.domInteractive = Number(new Date())
-    spans = createNavigationTimingSpans(timings, timings.fetchStart)
-    expect(spans.map(mapSpan)).toEqual([
-      { name: 'Fire "DOMContentLoaded" event', _end: 1498, _start: 1464 },
-      { name: 'Fire "load" event', _end: 2874, _start: 2852 }
-    ])
+    const domInteractiveValues = [undefined, null, 0, 1, Number(new Date())]
+    for (let i = 0; i < domInteractiveValues.length; i++) {
+      const value = domInteractiveValues[i]
+      const timingObj = {
+        ...timings,
+        domInteractive: value,
+        requestStart: 0
+      }
+      const spans = createNavigationTimingSpans(timingObj, transactionEnd)
+      expect(spans.map(mapSpan)).toEqual([
+        { name: 'Fire "DOMContentLoaded" event', _end: 1498, _start: 1464 },
+        { name: 'Fire "load" event', _end: 2874, _start: 2852 }
+      ])
+    }
 
-    timings.domContentLoadedEventStart = 'testing'
-    timings.domContentLoadedEventEnd = 'testings'
-    spans = createNavigationTimingSpans(timings, timings.fetchStart)
+    const timingsObj = {
+      ...timings,
+      domInteractive: 0,
+      requestStart: 0,
+      domContentLoadedEventStart: 'a'
+    }
+    spans = createNavigationTimingSpans(timingsObj, transactionEnd)
     expect(spans.map(mapSpan)).toEqual([
       { name: 'Fire "load" event', _end: 2874, _start: 2852 }
     ])
-    // console.log(spans.map(s => `${s._start}, ${s._end}, ${s.duration()}, ${s.name}`).join('\n'))
   })
 
   it('should createResourceTimingSpans', function() {
-    const spans = createResourceTimingSpans(resourceEntries, [
-      'http://ajax-filter.test'
-    ])
+    const spans = createResourceTimingSpans(
+      resourceEntries,
+      ['http://ajax-filter.test'],
+      transactionEnd
+    )
     expect(spans.map(mapSpan)).toEqual(spanSnapshot)
   })
 
   it('should captureHardNavigation', function() {
     var tr = new Transaction('test', 'test')
     tr.isHardNavigation = true
     tr.end()
+    /**
+     * In an ideal scenaio, hard navigation happens only after page
+     * load event has fired
+     */
+    tr._end = performance.timing.loadEventEnd
```

Comment:
the whole idea of this to keep the test in a predictable way, since we are closing the transaction immediately, transaction end value will be much lesser and the check for `expect(tr.spans.length).toBeGreaterThan(1)` would fail because all `spans end` will be more than transaction end which would make all spans invalid.

Reply:
I will remove this for now, but we can see the effect later on.